File: src/main/java/kr/hailor/hailor/entity/Designer.java
package kr.hailor.hailor.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "designers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Designer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 디자이너 이름 (예: "이초 디자이너")
    @Column(nullable = false)
    private String name;

    // 헤어샵 주소 (예: "서울 강남구 압구정로79길")
    private String hairShopAddress;

    // 지역 (예: "강남/청담/압구정")
    private String region;

    // 전문 분야 (예: "펌")
    private String specialty;

    // 대면 컨설팅 금액 (예: 40000)
    private int offlinePrice;

    // 비대면 컨설팅 금액 (예: 20000)
    private int onlinePrice;

    // 대면/비대면 옵션 (예: "대면, 비대면")
    private String consultationOption;

    // 한 줄 소개 (띄어쓰기 포함 최대 30자)
    @Column(length = 30)
    private String shortDescription;

    // 추가: 프로필 사진 URL
    private String profilePicture;

    // 대면 선택 시 상세 메시지 (상수로 정의하여 DTO 전달 시 사용)
    public static final String OFFLINE_CONSULTING_MESSAGE =
            "30,000₩ 부터 시작\n실제 샵에 방문하여 컨설팅 진행";

    // 화상(비대면) 선택 시 상세 메시지
    public static final String ONLINE_CONSULTING_MESSAGE =
            "20,000₩ 부터 시작\n예약 완료 후 생성되는 구글미트에서 화상으로 컨설팅 진행";

    // 공통 컨설팅 안내 메시지
    public static final String COMMON_CONSULTING_MESSAGE =
            "소요시간 약 30분 진행\n컨설팅 내용은 진행 후 요약된 리포트 고객에게 전달됩니다";
}

File: src/main/java/kr/hailor/hailor/enity/Designer.kt
package kr.hailor.hailor.enity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table

@Entity
@Table(name = "designers")
@SequenceGenerator(name = "DESIGNER_SEQ_GENERATOR", sequenceName = "DESIGNER_SEQ", initialValue = 1, allocationSize = 1)
class Designer(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "DESIGNER_SEQ_GENERATOR")
    @Column(updatable = false, nullable = false)
    val id: Long = 0L,
    @Column(length = 20, nullable = false)
    val name: String,
    @Column(length = 255, nullable = false)
    val shopAddress: String,
    @ManyToOne
    val region: DesignerRegion,
    @Enumerated(EnumType.STRING)
    @Column(length = 50, nullable = false)
    val specialization: Specialization,
    @Enumerated(EnumType.STRING)
    @Column(length = 20, nullable = false)
    val meetingType: MeetingType,
    @Column(nullable = false)
    val offlinePrice: Int,
    @Column(nullable = false)
    val onlinePrice: Int,
    @Column(length = 255, nullable = false)
    val description: String,
    @Column(length = 50, nullable = false)
    val profileImageName: String,
) : BaseModifiableEntity()

enum class Specialization(
    val description: String,
) {
    PERM("펌"),
    DYEING("염색"),
    DYEING_BLEACHING("탈염색"),
}

enum class MeetingType(
    val description: String,
) {
    OFFLINE("대면"),
    ONLINE("비대면"),
    OFFLINE_AND_ONLINE("대면/비대면"),
}
